### ISSUE TITLE
docs: Rails 7.1 (unreleased) has automatic retries of retriable database errors

### ISSUE THREAD

**Issue by bf4** [tines/rails-pg-adapter#23]
https://github.com/rails/rails/blob/d026af43c35c25814b4d16511f1b8fc6d70a4569/activerecord/lib/active_record/connection_adapters/abstract_adapter.rb#L1028

**Comment by shayonj** [comment]
hi! Thanks for the report. Thats right, Rails 7 has retries. We have limited this gem to be Rails <=6.x currently (related: https://github.com/tines/rails-pg-adapter/pull/22)

Also, in case its useful for others, Rails 7 currently only allows retrying a select view read/begin queries, see here: https://github.com/search?q=repo%3Arails%2Frails%20allow_retry&type=code. Which makes sense, since Rails cannot assume to retry queries that were severed while the database was processing a request. See here (https://github.com/rails/rails/pull/48093). Which is something this gem does. The trade off here is that: The gem assumes its OK to retry those in-flight cancelled queries on users' behalf. 

Closing the issue, but let me know if there is some action you'd like for us to take. Thanks much again!
